scripts/create_php_tags.py

```diff
@@ -81,24 +77,25 @@ def parse_and_create_php_tags_file():
             # Also create a class tag when encountering a __construct()
             if tag_name == '__construct' and scope is not None:
                 scope, tag_name = split_scope(scope)
-                tag_list.append((tag_name, TYPE_CLASS, None, arg_list, scope))
+                tag_list.append((tag_name, TYPE_CLASS, None, arg_list, scope or ''))
 
     # write tags
     script_dir = dirname(__file__)
     tags_file_path = join(script_dir, '..', 'data', 'tags', 'std.php.tags')
-    with open(tags_file_path, 'w') as tags_file:
+    with open(tags_file_path, 'w', encoding='iso-8859-1') as tags_file:
```

Comment:
why ISO 8859-1? This seems fully ASCII currently, is this encoding documented somewhere?  I can't seem to find out anything in a response header or something.

Reply:
Isn't it the default encoding of PHP?  So its what the PHP symbols will be.  But tagmangler is a binary format, maybe it should be a binary file?

Reply:
The content parsed from the JSON is (probably) just ASCII. But the `TA_*` markers are not ASCII and are encoded in `iso-8859-1`, also for other files using this tagmanager format.

I just added it on writing the file to explicitly set it, as Python3 `open()` wants the `encoding` argument.

Reply:
Aah, it's the output file, right.  Well, OK then -- although I guess I'd open it binary (if it makes a difference)

Reply:
It does not. Tried it and the resulting file is identical.
In the generating script we had to use byte strings for writing to the binary yopened file and need to encode the generated tag line anyway.
So, I guess the remaining difference could be performance, maybe the one is 0.5ms faster than other. For me, the difference cannot be big enough to care if the script is executed every few years :).

Reply:
All the symbols are ASCII and tagmanager only uses a few low byte values (which are valid ASCII) IIRC so I guess the encoding won't matter for this purpose.

When PHP adds emoji to its symbols it might be different :stuck_out_tongue:

Reply:
> So, I guess the remaining difference could be performance, maybe the one is 0.5ms faster than other. For me, the difference cannot be big enough to care if the script is executed every few years :).

Nah, clearly that doesn't matter at all :)